src/test/scala/firrtlTests/CheckSpec.scala

```diff
@@ -282,4 +282,29 @@ class CheckSpec extends FlatSpec with Matchers {
       }
     }
   }
+
+  behavior of "Uniqueness"
+  CheckSpec.nonUniqueExamples.map{ case (description, input) =>
+    it should s"be asserted for $description" in {
+      assertThrows[CheckHighForm.NotUniqueException] {
+        Seq(ToWorkingIR, CheckHighForm).foldLeft(Parser.parse(input)){ case (c, tx) => tx.run(c) }
+      }
+    }
+  }
+}
+
+object CheckSpec {
+  val nonUniqueExamples =
+    Map("two ports with the same name"         -> """|circuit Top:
```

Comment:
Nit, I would just make this a `List`, with 3 entries the iteration order will always be the same but if there were more than 4 it would be indeterminate.

Reply:
Makes sense to me. I'm not using any `Map` features. `List` seems right.